vector-stores/spring-ai-milvus-store/src/main/java/org/springframework/ai/vectorstore/MilvusVectorStore.java

```diff
@@ -69,6 +69,8 @@ public class MilvusVectorStore implements VectorStore, InitializingBean {
 
 	public static final int INVALID_EMBEDDING_DIMENSION = -1;
 
+	private static final int MAX_EMBEDDING_ARRAY_DIMENSIONS = 2048;
```

Comment:
Is there a reason for this limit? Milvus documentation suggests much higher limits for dimensionality - https://milvus.io/docs/limitations.md#Dimensions-of-a-vector.

Reply:
The reason I added this dimensions limit is that when using OpenAiEmbeddingModel implementation of the EmbeddingModel is an OpenAi API limit:
[](https://platform.openai.com/docs/api-reference/embeddings/create)

I attached the link for the first version of the API, but there is the same situation for the second one!

Reply:
The reason I added this dimensions limit is that when using OpenAiEmbeddingModel implementation of the EmbeddingModel is an OpenAi API limit:
https://platform.openai.com/docs/api-reference/embeddings/create

I attached the link for the first version of the API, but there is the same situation for the second one!

Reply:
I renamed MAX_EMBEDDING_ARRAY_DIMENSIONS to the MAX_OPENAI_EMBEDDING_ARRAY_DIMENSIONS for greater clarity